libraries/chain/controller.cpp

```diff
@@ -3312,7 +3312,17 @@ struct controller_impl {
                transition_to_savanna_if_needed();
             }
 
-            if (!my_finalizers.empty()) {
+            bool try_voting = !my_finalizers.empty();
+#if 0
+            // this doesn't seem to work (transition to savanna stalls in cluster test), needs
+            // further investigation [greg]
+            if (s == controller::block_status::complete || s == controller::block_status::validated) {
+               // if the block is complete or validated, we should already have tried voting on it
+               try_voting = false;
+            }
+#endif
+            if (try_voting) {
+               // if the block is complete or validated, we have already voted on it
```

Comment:
Would like to prevent pushing this in until we know what is going on. Just revert this for now.

Reply:
done.